pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -209,6 +215,12 @@ func (cb *ClusterBuilder) applyDestinationRule(mc *MutableCluster, clusterMode C
 	return subsetClusters
 }
 
+func (cb *ClusterBuilder) applyMetadataExchange(c *cluster.Cluster) {
+	if cb.push.Telemetry.AnyTelemetryExists() {
```

Comment:
where does it makes sense to document this coupling?  should we put this in the API docs?

Reply:
I think its likely a short term "feature flag" enablement of sorts. Long term we will probably have a more graceful migration from envoyfilter to Telemetry API.

So probably wherever we document Telemetry as experimental, we mention how this works and that you need to disable EnvoyFitler to enable Telemetry